File: app/schemas/study.py
from pydantic import BaseModel, Field, field_validator
from datetime import datetime
from typing import Any

from app.schemas.user import strip_whitespace


class StudyGuideCreate(BaseModel):
    """Request to generate a study guide."""
    assignment_id: int | None = None
    course_id: int | None = None
    course_content_id: int | None = None
    title: str | None = Field(default=None, max_length=200)  # Optional custom title
    content: str | None = Field(default=None, max_length=50000)  # Optional custom content to base guide on
    regenerate_from_id: int | None = None  # ID of existing guide to create new version of
    custom_prompt: str | None = Field(default=None, max_length=5000)  # Custom AI prompt (for "Other" tool selection)
    focus_prompt: str | None = Field(default=None, max_length=2000)  # Optional focus area for AI generation
    document_type: str | None = Field(default=None, max_length=30)  # Auto-detected or user-selected document type (§6.105)
    study_goal: str | None = Field(default=None, max_length=30)  # Study goal selection (§6.105)
    study_goal_text: str | None = Field(default=None, max_length=200)  # Free-text study goal details (§6.105)

    @field_validator('title', 'custom_prompt', 'focus_prompt', mode='before')
    @classmethod
    def _strip_whitespace(cls, v: object) -> object:
        return strip_whitespace(v)


class AutoCreatedTask(BaseModel):
    """Summary of an auto-created task."""
    id: int
    title: str
    due_date: str
    priority: str


class StudyGuideResponse(BaseModel):
    """Study guide response."""
    id: int
    user_id: int
    assignment_id: int | None
    course_id: int | None
    course_content_id: int | None = None
    title: str
    content: str
    guide_type: str
    version: int = 1
    parent_guide_id: int | None = None
    relationship_type: str | None = None
    generation_context: str | None = None
    focus_prompt: str | None = None
    is_truncated: bool = False
    parent_summary: str | None = None
    curriculum_codes: str | None = None  # JSON string
    suggestion_topics: str | None = None
    answer_key_markdown: str | None = None
    weak_topics: str | None = None
    ai_engine: str | None = None
    created_at: datetime
    archived_at: datetime | None = None
    safety_checked: bool = True
    auto_created_tasks: list[AutoCreatedTask] = []

    # Sharing fields
    shared_with_user_id: int | None = None
    shared_at: datetime | None = None
    viewed_at: datetime | None = None
    viewed_count: int = 0
    shared_with_name: str | None = None

    class Config:
        from_attributes = True


class AnswerKeyResponse(BaseModel):
    """Response after generating an answer key for a worksheet."""
    id: int
    title: str
    guide_type: str
    answer_key_markdown: str | None = None
    created_at: datetime

    class Config:
        from_attributes = True


class QuizGenerateRequest(BaseModel):
    """Request to generate a quiz."""
    assignment_id: int | None = None
    course_id: int | None = None
    course_content_id: int | None = None
    topic: str | None = Field(default=None, max_length=200)
    content: str | None = Field(default=None, max_length=50000)
    num_questions: int = Field(default=5, ge=1, le=50)
    regenerate_from_id: int | None = None
    focus_prompt: str | None = Field(default=None, max_length=2000)  # Optional focus area for AI generation
    difficulty: str | None = Field(default=None, max_length=10)  # easy, medium, hard
    document_type: str | None = Field(default=None, max_length=30)  # §6.106: inherited on regeneration
    study_goal: str | None = Field(default=None, max_length=30)
    study_goal_text: str | None = Field(default=None, max_length=200)

    @field_validator('topic', 'focus_prompt', mode='before')
    @classmethod
    def _strip_whitespace(cls, v: object) -> object:
        return strip_whitespace(v)

    @field_validator('difficulty', mode='before')
    @classmethod
    def _validate_difficulty(cls, v: object) -> object:
        if v is None:
            return None
        s = str(v).lower().strip()
        if s not in ('easy', 'medium', 'hard'):
            raise ValueError('difficulty must be easy, medium, or hard')
        return s


class QuizQuestion(BaseModel):
    """A single quiz question."""
    question: str
    options: dict[str, str]  # {"A": "...", "B": "...", ...}
    correct_answer: str
    explanation: str


class QuizResponse(BaseModel):
    """Quiz response with questions."""
    id: int
    title: str
    questions: list[QuizQuestion]
    guide_type: str = "quiz"
    course_content_id: int | None = None
    version: int = 1
    parent_guide_id: int | None = None
    created_at: datetime
    safety_checked: bool = True
    auto_created_tasks: list[AutoCreatedTask] = []

    class Config:
        from_attributes = True


class FlashcardGenerateRequest(BaseModel):
    """Request to generate flashcards."""
    assignment_id: int | None = None
    course_id: int | None = None
    course_content_id: int | None = None
    topic: str | None = Field(default=None, max_length=200)
    content: str | None = Field(default=None, max_length=50000)
    num_cards: int = Field(default=10, ge=1, le=50)
    regenerate_from_id: int | None = None
    focus_prompt: str | None = Field(default=None, max_length=2000)  # Optional focus area for AI generation
    document_type: str | None = Field(default=None, max_length=30)  # §6.106: inherited on regeneration
    study_goal: str | None = Field(default=None, max_length=30)
    study_goal_text: str | None = Field(default=None, max_length=200)

    @field_validator('topic', 'focus_prompt', mode='before')
    @classmethod
    def _strip_whitespace(cls, v: object) -> object:
        return strip_whitespace(v)


class Flashcard(BaseModel):
    """A single flashcard."""
    front: str
    back: str


class FlashcardSetResponse(BaseModel):
    """Flashcard set response."""
    id: int
    title: str
    cards: list[Flashcard]
    guide_type: str = "flashcards"
    course_content_id: int | None = None
    version: int = 1
    parent_guide_id: int | None = None
    created_at: datetime
    safety_checked: bool = True
    auto_created_tasks: list[AutoCreatedTask] = []

    class Config:
        from_attributes = True


class MindMapGenerateRequest(BaseModel):
    """Request to generate a mind map."""
    assignment_id: int | None = None
    course_id: int | None = None
    course_content_id: int | None = None
    topic: str | None = Field(default=None, max_length=200)
    content: str | None = Field(default=None, max_length=50000)
    regenerate_from_id: int | None = None
    focus_prompt: str | None = Field(default=None, max_length=2000)

    @field_validator('topic', 'focus_prompt', mode='before')
    @classmethod
    def _strip_whitespace(cls, v: object) -> object:
        return strip_whitespace(v)


class MindMapBranch(BaseModel):
    """A single mind map branch child."""
    label: str
    detail: str = ""


class MindMapBranchGroup(BaseModel):
    """A top-level mind map branch with children."""
    label: str
    children: list[MindMapBranch] = []


class MindMapData(BaseModel):
    """The mind map data structure."""
    central_topic: str
    branches: list[MindMapBranchGroup] = []


class MindMapResponse(BaseModel):
    """Mind map response."""
    id: int
    title: str
    mind_map: MindMapData
    guide_type: str = "mind_map"
    version: int = 1
    parent_guide_id: int | None = None
    created_at: datetime
    safety_checked: bool = True
    auto_created_tasks: list[AutoCreatedTask] = []

    class Config:
        from_attributes = True


class StudyGuideUpdate(BaseModel):
    """Request to update a study guide (e.g. assign/categorize to a course, rename)."""
    title: str | None = Field(default=None, max_length=200)
    course_id: int | None = None
    course_content_id: int | None = None

    @field_validator('title', mode='before')
    @classmethod
    def _strip_whitespace(cls, v: object) -> object:
        return strip_whitespace(v)


class GenerateChildRequest(BaseModel):
    """Request to generate a child sub-guide from selected text."""
    topic: str = Field(min_length=3, max_length=5000)  # The selected text
    guide_type: str = Field(default="study_guide", max_length=50)  # study_guide, quiz, flashcards, worksheet, weak_area_analysis, high_level_summary, answer_key, problem_solver
    custom_prompt: str | None = Field(default=None, max_length=2000)  # Optional focus (e.g., "make it harder")
    # §6.106: Strategy context inherited from parent
    document_type: str | None = Field(default=None, max_length=30)
    study_goal: str | None = Field(default=None, max_length=30)
    study_goal_text: str | None = Field(default=None, max_length=200)
    max_tokens: int | None = Field(default=None, ge=500, le=4000)

    @field_validator('guide_type')
    @classmethod
    def validate_guide_type(cls, v: str) -> str:
        if v not in ('study_guide', 'quiz', 'flashcards', 'worksheet', 'weak_area_analysis', 'high_level_summary', 'answer_key', 'problem_solver'):
            raise ValueError('guide_type must be study_guide, quiz, flashcards, worksheet, weak_area_analysis, high_level_summary, answer_key, or problem_solver')
        return v

    @field_validator('topic', 'custom_prompt', mode='before')
    @classmethod
    def _strip_whitespace(cls, v: object) -> object:
        return strip_whitespace(v)


class DuplicateCheckRequest(BaseModel):
    """Request to check for duplicate study guide before generation."""
    title: str | None = Field(default=None, max_length=200)
    guide_type: str = Field(max_length=50)  # study_guide, quiz, flashcards
    assignment_id: int | None = None
    course_id: int | None = None

    @field_validator('title', mode='before')
    @classmethod
    def _strip_whitespace(cls, v: object) -> object:
        return strip_whitespace(v)


class WeakAreaAnalyzeRequest(BaseModel):
    """Request to analyze weak areas from a test/exam."""
    content_id: int
    student_id: int | None = None


class StudyGuideTreeNode(BaseModel):
    """A node in the study guide tree hierarchy."""
    id: int
    title: str
    guide_type: str
    created_at: datetime
    children: list["StudyGuideTreeNode"] = []

    class Config:
        from_attributes = True


class StudyGuideTreeResponse(BaseModel):
    """Full tree hierarchy for a study guide with breadcrumb path."""
    root: StudyGuideTreeNode
    current_path: list[int]  # IDs from root to current guide


class DuplicateCheckResponse(BaseModel):
    """Response indicating whether a duplicate study guide exists."""
    exists: bool
    existing_guide: StudyGuideResponse | None = None
    message: str | None = None


# §6.114 — Study Q&A save actions
class SaveQAAsGuideRequest(BaseModel):
    """Save a Q&A response as a sub-guide."""
    content: str = Field(..., min_length=1)
    title: str = Field(default="", max_length=255)


class SaveQAAsMaterialRequest(BaseModel):
    """Save a Q&A response as a course material."""
    content: str = Field(..., min_length=1)
    title: str = Field(default="", max_length=255)


class SaveAsMaterialResponse(BaseModel):
    """Response from saving a Q&A response as course material."""
    id: int
    title: str
    course_id: int
    content_type: str

    class Config:
        from_attributes = True


class ClassifyDocumentResponse(BaseModel):
    """Response from document classification endpoint."""
    document_type: str
    confidence: float
    detected_subject: str = "other"
    subject_keywords_found: list[str] = []
    material_type_display: str = "Custom Document"
    error: bool = False


# ── Worksheet generation (#2956) ──

WORKSHEET_TEMPLATES = {
    "worksheet_general",
    "worksheet_math_word_problems",
    "worksheet_english",
    "worksheet_french",
}

WORKSHEET_DIFFICULTIES = {"below_grade", "grade_level", "above_grade"}


class WorksheetGenerateRequest(BaseModel):
    """Request to generate a worksheet."""
    content_id: int
    template_key: str = Field(default="worksheet_general", max_length=50)
    num_questions: int = Field(default=10, ge=5, le=20)
    difficulty: str = Field(default="grade_level", max_length=20)
    student_id: int | None = None

    @field_validator('template_key')
    @classmethod
    def validate_template_key(cls, v: str) -> str:
        if v not in WORKSHEET_TEMPLATES:
            raise ValueError(f"template_key must be one of: {', '.join(sorted(WORKSHEET_TEMPLATES))}")
        return v

    @field_validator('difficulty')
    @classmethod
    def validate_difficulty(cls, v: str) -> str:
        if v not in WORKSHEET_DIFFICULTIES:
            raise ValueError(f"difficulty must be one of: {', '.join(sorted(WORKSHEET_DIFFICULTIES))}")
        return v


class WorksheetResponse(BaseModel):
    """Worksheet response."""
    id: int
    user_id: int
    course_id: int | None
    course_content_id: int | None
    title: str
    content: str
    guide_type: str = "worksheet"
    template_key: str | None = None
    num_questions: int | None = None
    difficulty: str | None = None
    answer_key_markdown: str | None = None
    created_at: datetime
    auto_created_tasks: list[AutoCreatedTask] = []

    class Config:
        from_attributes = True

File: frontend/src/api/study.ts
import { api, AI_TIMEOUT } from './client';

// Study Guide Types
export interface AutoCreatedTask {
  id: number;
  title: string;
  due_date: string;
  priority: string;
}

export interface StudyGuide {
  id: number;
  user_id: number;
  assignment_id: number | null;
  course_id: number | null;
  course_content_id: number | null;
  title: string;
  content: string;
  guide_type: string;
  version: number;
  parent_guide_id: number | null;
  relationship_type?: string | null;
  generation_context?: string | null;
  focus_prompt: string | null;
  is_truncated?: boolean;
  parent_summary?: string | null;
  created_at: string;
  archived_at: string | null;
  auto_created_tasks?: AutoCreatedTask[];
  // Sharing fields
  shared_with_user_id?: number | null;
  shared_at?: string | null;
  viewed_at?: string | null;
  viewed_count?: number;
  shared_with_name?: string | null;
  // §6.106: Strategy context
  document_type?: string | null;
  study_goal?: string | null;
  study_goal_text?: string | null;
  suggestion_topics?: string | null;
}

// Sharing types
export interface SharedGuideStatus {
  id: number;
  title: string;
  guide_type: string;
  shared_with_user_id: number | null;
  shared_with_name: string | null;
  shared_at: string | null;
  viewed_at: string | null;
  viewed_count: number;
  status: 'not_shared' | 'shared' | 'viewed';
  created_at: string;
}

export interface SharedWithMeGuide {
  id: number;
  title: string;
  content: string;
  guide_type: string;
  shared_by_name: string;
  shared_at: string;
  viewed_at: string | null;
  viewed_count: number;
  created_at: string;
  course_content_id: number | null;
}

export interface StudyGuideTreeNode {
  id: number;
  title: string;
  guide_type: string;
  created_at: string;
  children: StudyGuideTreeNode[];
}

export interface StudyGuideTreeResponse {
  root: StudyGuideTreeNode;
  current_path: number[];
}

export interface DuplicateCheckResponse {
  exists: boolean;
  existing_guide: StudyGuide | null;
  message: string | null;
}

export interface QuizQuestion {
  question: string;
  options: { A: string; B: string; C: string; D: string };
  correct_answer: string;
  explanation: string;
}

export interface Quiz {
  id: number;
  title: string;
  questions: QuizQuestion[];
  guide_type: string;
  course_content_id: number | null;
  version: number;
  parent_guide_id: number | null;
  created_at: string;
  auto_created_tasks?: AutoCreatedTask[];
}

export interface Flashcard {
  front: string;
  back: string;
}

export interface FlashcardSet {
  id: number;
  title: string;
  cards: Flashcard[];
  guide_type: string;
  course_content_id: number | null;
  version: number;
  parent_guide_id: number | null;
  created_at: string;
  auto_created_tasks?: AutoCreatedTask[];
}

export interface MindMapBranch {
  label: string;
  detail: string;
}

export interface MindMapBranchGroup {
  label: string;
  children: MindMapBranch[];
}

export interface MindMapData {
  central_topic: string;
  branches: MindMapBranchGroup[];
}

export interface MindMap {
  id: number;
  title: string;
  mind_map: MindMapData;
  guide_type: string;
  version: number;
  parent_guide_id: number | null;
  created_at: string;
  auto_created_tasks?: AutoCreatedTask[];
}

export interface SupportedFormats {
  documents: string[];
  spreadsheets: string[];
  presentations: string[];
  images: string[];
  archives: string[];
  max_file_size_mb: number;
  ocr_available: boolean;
}

export interface ExtractedText {
  filename: string;
  text: string;
  character_count: number;
  word_count: number;
}

// Quiz Result Types
export interface QuizResultCreate {
  study_guide_id: number;
  score: number;
  total_questions: number;
  answers: Record<number, string>;
  time_taken_seconds?: number;
  student_user_id?: number;
}

export interface QuizResultResponse {
  id: number;
  user_id: number;
  study_guide_id: number;
  score: number;
  total_questions: number;
  percentage: number;
  answers_json: string;
  attempt_number: number;
  time_taken_seconds: number | null;
  completed_at: string;
  quiz_title: string | null;
}

export interface QuizResultSummary {
  id: number;
  study_guide_id: number;
  quiz_title: string | null;
  score: number;
  total_questions: number;
  percentage: number;
  attempt_number: number;
  completed_at: string;
}

export interface QuizHistoryStats {
  total_attempts: number;
  unique_quizzes: number;
  average_score: number;
  best_score: number;
  recent_trend: 'improving' | 'declining' | 'stable';
}

export interface ResolvedStudent {
  student_user_id: number;
  student_name: string;
}

export async function classifyDocument(textContent: string, filename: string): Promise<{ document_type: string; confidence: number }> {
  const response = await api.post('/api/study/classify-document', {
    text_content: textContent,
    filename,
  });
  return response.data;
}

// Study Tools API
export const studyApi = {
  generateGuide: async (params: { assignment_id?: number; course_id?: number; course_content_id?: number; title?: string; content?: string; regenerate_from_id?: number; custom_prompt?: string; focus_prompt?: string; document_type?: string; study_goal?: string; study_goal_text?: string }) => {
    const response = await api.post('/api/study/generate', params, AI_TIMEOUT);
    return response.data as StudyGuide;
  },

  generateQuiz: async (params: { assignment_id?: number; course_id?: number; course_content_id?: number; topic?: string; content?: string; num_questions?: number; regenerate_from_id?: number; focus_prompt?: string; difficulty?: string; document_type?: string; study_goal?: string; study_goal_text?: string }) => {
    const response = await api.post('/api/study/quiz/generate', params, AI_TIMEOUT);
    return response.data as Quiz;
  },

  generateFlashcards: async (params: { assignment_id?: number; course_id?: number; course_content_id?: number; topic?: string; content?: string; num_cards?: number; regenerate_from_id?: number; focus_prompt?: string; document_type?: string; study_goal?: string; study_goal_text?: string }) => {
    const response = await api.post('/api/study/flashcards/generate', params, AI_TIMEOUT);
    return response.data as FlashcardSet;
  },

  generateMindMap: async (params: { assignment_id?: number; course_id?: number; course_content_id?: number; topic?: string; content?: string; regenerate_from_id?: number; focus_prompt?: string }) => {
    const response = await api.post('/api/study/mind-map/generate', params, AI_TIMEOUT);
    return response.data as MindMap;
  },

  checkDuplicate: async (params: { title?: string; guide_type: string; assignment_id?: number; course_id?: number }) => {
    const response = await api.post('/api/study/check-duplicate', params);
    return response.data as DuplicateCheckResponse;
  },

  listGuides: async (params?: { guide_type?: string; course_id?: number; course_content_id?: number; include_children?: boolean; include_archived?: boolean; student_user_id?: number }) => {
    const response = await api.get('/api/study/guides', { params: params || {} });
    return response.data as StudyGuide[];
  },

  listGuideVersions: async (guideId: number) => {
    const response = await api.get(`/api/study/guides/${guideId}/versions`);
    return response.data as StudyGuide[];
  },

  getGuide: async (id: number) => {
    const response = await api.get(`/api/study/guides/${id}`);
    return response.data as StudyGuide;
  },

  deleteGuide: async (id: number) => {
    await api.delete(`/api/study/guides/${id}`);
  },

  restoreGuide: async (id: number) => {
    const response = await api.patch(`/api/study/guides/${id}/restore`);
    return response.data as StudyGuide;
  },

  permanentDeleteGuide: async (id: number) => {
    await api.delete(`/api/study/guides/${id}/permanent`);
  },

  updateGuide: async (id: number, data: { title?: string; course_id?: number | null; course_content_id?: number | null }) => {
    const response = await api.patch(`/api/study/guides/${id}`, data);
    return response.data as StudyGuide;
  },

  continueGuide: async (guideId: number) => {
    const response = await api.post(`/api/study/${guideId}/continue`, null, AI_TIMEOUT);
    return response.data as StudyGuide;
  },

  /** Return the SSE endpoint URL for streaming continuation of a truncated guide. */
  continueGuideStreamUrl: (guideId: number) => `/api/study/${guideId}/continue-stream`,

  // File Upload Methods
  getSupportedFormats: async () => {
    const response = await api.get('/api/study/upload/formats');
    return response.data as SupportedFormats;
  },

  generateFromFile: async (params: {
    file: File;
    title?: string;
    guide_type: 'study_guide' | 'quiz' | 'flashcards';
    num_questions?: number;
    num_cards?: number;
    course_id?: number;
    course_content_id?: number;
    focus_prompt?: string;
    difficulty?: string;
    document_type?: string;
    study_goal?: string;
    study_goal_text?: string;
  }) => {
    const formData = new FormData();
    formData.append('file', params.file);
    if (params.title) formData.append('title', params.title);
    formData.append('guide_type', params.guide_type);
    if (params.num_questions) formData.append('num_questions', params.num_questions.toString());
    if (params.num_cards) formData.append('num_cards', params.num_cards.toString());
    if (params.course_id) formData.append('course_id', params.course_id.toString());
    if (params.course_content_id) formData.append('course_content_id', params.course_content_id.toString());
    if (params.focus_prompt) formData.append('focus_prompt', params.focus_prompt);
    if (params.difficulty) formData.append('difficulty', params.difficulty);
    if (params.document_type) formData.append('document_type', params.document_type);
    if (params.study_goal) formData.append('study_goal', params.study_goal);
    if (params.study_goal_text) formData.append('study_goal_text', params.study_goal_text);

    const response = await api.post('/api/study/upload/generate', formData, AI_TIMEOUT);
    return response.data as StudyGuide;
  },

  generateFromTextAndImages: async (params: {
    content: string;
    images: File[];
    title?: string;
    guide_type: 'study_guide' | 'quiz' | 'flashcards';
    num_questions?: number;
    num_cards?: number;
    course_id?: number;
    course_content_id?: number;
    focus_prompt?: string;
    difficulty?: string;
    document_type?: string;
    study_goal?: string;
    study_goal_text?: string;
  }) => {
    const formData = new FormData();
    formData.append('content', params.content);
    if (params.title) formData.append('title', params.title);
    formData.append('guide_type', params.guide_type);
    if (params.num_questions) formData.append('num_questions', params.num_questions.toString());
    if (params.num_cards) formData.append('num_cards', params.num_cards.toString());
    if (params.course_id) formData.append('course_id', params.course_id.toString());
    if (params.course_content_id) formData.append('course_content_id', params.course_content_id.toString());
    if (params.focus_prompt) formData.append('focus_prompt', params.focus_prompt);
    if (params.difficulty) formData.append('difficulty', params.difficulty);
    if (params.document_type) formData.append('document_type', params.document_type);
    if (params.study_goal) formData.append('study_goal', params.study_goal);
    if (params.study_goal_text) formData.append('study_goal_text', params.study_goal_text);
    params.images.forEach(img => formData.append('images', img));

    const response = await api.post('/api/study/generate-with-images', formData, AI_TIMEOUT);
    return response.data as StudyGuide;
  },

  extractTextFromFile: async (file: File) => {
    const formData = new FormData();
    formData.append('file', file);

    const response = await api.post('/api/study/upload/extract-text', formData, AI_TIMEOUT);
    return response.data as ExtractedText;
  },

  // Quiz Results Methods
  saveQuizResult: async (data: QuizResultCreate) => {
    const response = await api.post('/api/quiz-results/', data);
    return response.data as QuizResultResponse;
  },

  getQuizHistory: async (params?: { study_guide_id?: number; student_user_id?: number; limit?: number; offset?: number }) => {
    const response = await api.get('/api/quiz-results/', { params: params || {} });
    return response.data as QuizResultSummary[];
  },

  getQuizStats: async (params?: { student_user_id?: number }) => {
    const response = await api.get('/api/quiz-results/stats', { params: params || {} });
    return response.data as QuizHistoryStats;
  },

  getQuizResult: async (id: number) => {
    const response = await api.get(`/api/quiz-results/${id}`);
    return response.data as QuizResultResponse;
  },

  deleteQuizResult: async (id: number) => {
    await api.delete(`/api/quiz-results/${id}`);
  },

  generateChildGuide: async (guideId: number, params: { topic: string; guide_type: string; custom_prompt?: string; document_type?: string; study_goal?: string; max_tokens?: number }) => {
    const response = await api.post(`/api/study/guides/${guideId}/generate-child`, params, AI_TIMEOUT);
    return response.data as StudyGuide;
  },

  listChildGuides: async (guideId: number) => {
    const response = await api.get(`/api/study/guides/${guideId}/children`);
    return response.data as StudyGuide[];
  },

  getGuideTree: async (guideId: number) => {
    const response = await api.get(`/api/study/guides/${guideId}/tree`);
    return response.data as StudyGuideTreeResponse;
  },

  resolveStudent: async (params: { course_id?: number; study_guide_id?: number }) => {
    const response = await api.get('/api/quiz-results/resolve-student', { params });
    return response.data as ResolvedStudent | null;
  },

  // Study Sharing (Parent-Child Study Link #1414)
  shareGuide: async (guideId: number, studentId: number) => {
    const response = await api.post(`/api/study-guides/${guideId}/share`, { student_id: studentId });
    return response.data;
  },

  getSharedWithMe: async () => {
    const response = await api.get('/api/study-guides/shared-with-me');
    return response.data as SharedWithMeGuide[];
  },

  markViewed: async (guideId: number) => {
    const response = await api.post(`/api/study-guides/${guideId}/mark-viewed`);
    return response.data;
  },

  getSharedStatus: async () => {
    const response = await api.get('/api/study-guides/shared-status');
    return response.data as SharedGuideStatus[];
  },
};
